### DISCUSSION TITLE
Segfault writing out file

### DISCUSSION THREAD

**Discussion by JJHelly** [pyswmm/pyswmm#532] [Q&A]
Aloha.  Iʻm modifying the Latte_art example to use my model.  The model runs but when I try to execute:

with Output('./Example1.out') as out:
     node_head_outfile = out.node_series('POC-1', NodeAttribute.TOTAL_INFLOW)

the script segfaults.  It runs fine for the Latte_art example, so I suspect it has to do with my node definitions but, being a relative newb, not sure how to debug this or what to look for.

Any suggestions would be very welcome.

**Comment by bemcdonnell** [comment]
@JJHelly sorry for the delay - in the INP can you confirm you are reporting all of the values:

```
[REPORT]
;;Reporting Options
INPUT      YES
CONTROLS   YES
AVERAGES   YES
SUBCATCHMENTS ALL
NODES ALL
LINKS ALL
```

**Reply by bemcdonnell** [reply]
@JJHelly just to confirm you checked this in the INP? And the node names are case sensitive too

**Comment by JJHelly** [comment]
Aloha.

I set the options as below but still fails apparently at:

with Output('./3086_Post_Alt1.inp.JJH.out') as out:
     node_head_outfile = out.node_series('POC-1', 
NodeAttribute.HYDRAULIC_HEAD)

Not sure how to examine the NodeAttributes in this situation.  Any 
advice on that would be helpful.

J.

On 5/23/24 05:36, Bryant E. McDonnell wrote:
>
> @JJHelly 
> <https://urldefense.com/v3/__https://github.com/JJHelly__;!!Mih3wA!Amizb2cbh6Jp8A2hfL0gWifPU7GD5IhhxqZtZVoInNXFeS8tYMT7_maYe4K4WP0g1-1m-tbq0l6lGeDzgwojDyoY$> 
> sorry for the delay - in the INP can you confirm you are reporting all 
> of the values:
>
> |[REPORT] ;;Reporting Options INPUT YES CONTROLS YES AVERAGES YES 
> SUBCATCHMENTS ALL NODES ALL LINKS ALL |
>
> —
> Reply to this email directly, view it on GitHub 
> <https://urldefense.com/v3/__https://github.com/pyswmm/pyswmm/discussions/532*discussioncomment-9536736__;Iw!!Mih3wA!Amizb2cbh6Jp8A2hfL0gWifPU7GD5IhhxqZtZVoInNXFeS8tYMT7_maYe4K4WP0g1-1m-tbq0l6lGeDzg-vAXVNK$>, 
> or unsubscribe 
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AZ3YXQGFTWKAP2CMKDQA4SLZDYEJDAVCNFSM6AAAAABICW7OXCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TKMZWG4ZTM__;!!Mih3wA!Amizb2cbh6Jp8A2hfL0gWifPU7GD5IhhxqZtZVoInNXFeS8tYMT7_maYe4K4WP0g1-1m-tbq0l6lGeDzg0MKyW4G$>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

-- 
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj  / 808 205 9882 / 760 8408660

--------------96TenHa0x6LuVgguXiXXQgGx
Content-Type: text/html; charset=UTF-8
Content-Transfer-Encoding: 8bit

<!DOCTYPE html><html><head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
  </head>
  <body>
    Aloha.<br>
    <br>
    I set the options as below but still fails apparently at:<br>
    <br>
    <font face="Courier New, Courier, monospace">with
      Output('./3086_Post_Alt1.inp.JJH.out') as out:<br>
      &nbsp;&nbsp;&nbsp; node_head_outfile = out.node_series('POC-1',&nbsp;&nbsp;
      NodeAttribute.HYDRAULIC_HEAD)<br>
      <br>
    </font>Not sure how to examine the NodeAttributes in this
    situation.&nbsp; Any advice on that would be helpful. <br>
    <br>
    J.<br>
    <br>
    <div class="moz-cite-prefix">On 5/23/24 05:36, Bryant E. McDonnell
      wrote:<br>
    </div>
    <blockquote type="cite" ***@***.***">
      
      <p dir="auto"><a class="user-mention notranslate" data-hovercard-type="user" data-hovercard-url="/users/JJHelly/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://urldefense.com/v3/__https://github.com/JJHelly__;!!Mih3wA!Amizb2cbh6Jp8A2hfL0gWifPU7GD5IhhxqZtZVoInNXFeS8tYMT7_maYe4K4WP0g1-1m-tbq0l6lGeDzgwojDyoY$" ***@***.***</a> sorry for the delay - in
        the INP can you confirm you are reporting all of the values:</p>
      <pre class="notranslate"><code class="notranslate">[REPORT]
;;Reporting Options
INPUT      YES
CONTROLS   YES
AVERAGES   YES
SUBCATCHMENTS ALL
NODES ALL
LINKS ALL
</code></pre>
      <p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">—<br>
        Reply to this email directly, <a href="https://urldefense.com/v3/__https://github.com/pyswmm/pyswmm/discussions/532*discussioncomment-9536736__;Iw!!Mih3wA!Amizb2cbh6Jp8A2hfL0gWifPU7GD5IhhxqZtZVoInNXFeS8tYMT7_maYe4K4WP0g1-1m-tbq0l6lGeDzg-vAXVNK$" moz-do-not-send="true">view it on GitHub</a>, or <a href="https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AZ3YXQGFTWKAP2CMKDQA4SLZDYEJDAVCNFSM6AAAAABICW7OXCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TKMZWG4ZTM__;!!Mih3wA!Amizb2cbh6Jp8A2hfL0gWifPU7GD5IhhxqZtZVoInNXFeS8tYMT7_maYe4K4WP0g1-1m-tbq0l6lGeDzg0MKyW4G$" moz-do-not-send="true">unsubscribe</a>.<br>
        You are receiving this because you were mentioned.<img src="https://github.com/notifications/beacon/AZ3YXQEMSVYAWR5LHBPG4ILZDYEJDA5CNFSM6AAAAABICW7OXCWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3TUL5UWJTQASGCOA.gif" alt="" moz-do-not-send="true" width="1" height="1"><span style="color: transparent; font-size: 0; display: none; visibility: hidden; overflow: hidden; opacity: 0; width: 0; height: 0; max-width: 0; max-height: 0; mso-hide: all">Message
          ID: <span>&lt;pyswmm/pyswmm/repo-discussions/532/comments/9536736</span><span>@</span><span>github</span><span>.</span><span>com&gt;</span></span></p>
      <script type="application/ld+json">[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/pyswmm/pyswmm/discussions/532#discussioncomment-9536736",
"url": "https://github.com/pyswmm/pyswmm/discussions/532#discussioncomment-9536736",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]</script>
    </blockquote>
    <br>
    <pre class="moz-signature" cols="72">-- 
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography 
<a class="moz-txt-link-freetext" href="https://www.sdsc.edu/~hellyj">https://www.sdsc.edu/~hellyj</a> / 808 205 9882 / 760 8408660</pre>
  </body>
</html>

--------------96TenHa0x6LuVgguXiXXQgGx--

**Comment by JJHelly** [comment]
Yes, copied here for proof:

[REPORT]
;;Reporting Options
INPUT       YES
CONTROLS    YES
AVERAGES    YES
SUBCATCHMENTS ALL
NODES ALL
LINKS ALL

All node names are uppercase.

Mahalo.
J.


On 5/24/24 13:08, Bryant E. McDonnell wrote:
>
> @JJHelly 
> <https://urldefense.com/v3/__https://github.com/JJHelly__;!!Mih3wA!FHIGtRkF2o1Oh0lJL-jvSugEBRgh36PpR8sZ5kEQ9FSfzNKPXjiFtpYiDNJsq_3gtvnNMBa1naSofVjtMKLcUhrR$> 
> just to confirm you checked this in the INP? And the node names are 
> case sensitive too
>
> —
> Reply to this email directly, view it on GitHub 
> <https://urldefense.com/v3/__https://github.com/pyswmm/pyswmm/discussions/532*discussioncomment-9552104__;Iw!!Mih3wA!FHIGtRkF2o1Oh0lJL-jvSugEBRgh36PpR8sZ5kEQ9FSfzNKPXjiFtpYiDNJsq_3gtvnNMBa1naSofVjtMFxvnrnJ$>, 
> or unsubscribe 
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AZ3YXQDXOK5NOQPSAXXLM33ZD7CAPAVCNFSM6AAAAABICW7OXCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TKNJSGEYDI__;!!Mih3wA!FHIGtRkF2o1Oh0lJL-jvSugEBRgh36PpR8sZ5kEQ9FSfzNKPXjiFtpYiDNJsq_3gtvnNMBa1naSofVjtMGuSiFU9$>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

-- 
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj  / 808 205 9882 / 760 8408660

--------------SB4Sw5z6Wu3SRkcbfjCps8Lp
Content-Type: text/html; charset=UTF-8
Content-Transfer-Encoding: 8bit

<!DOCTYPE html><html><head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
  </head>
  <body>
    Yes, copied here for proof:<br>
    <br>
    <font face="Courier New, Courier, monospace">[REPORT]<br>
      ;;Reporting Options<br>
      INPUT&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; YES<br>
      CONTROLS&nbsp;&nbsp;&nbsp; YES<br>
      AVERAGES&nbsp;&nbsp;&nbsp; YES<br>
      SUBCATCHMENTS ALL<br>
      NODES ALL<br>
      LINKS ALL<br>
      <br>
      All node names are uppercase.<br>
      <br>
      Mahalo.<br>
      J.<br>
    </font><br>
    <br>
    <div class="moz-cite-prefix">On 5/24/24 13:08, Bryant E. McDonnell
      wrote:<br>
    </div>
    <blockquote type="cite" ***@***.***">
      
      <p dir="auto"><a class="user-mention notranslate" data-hovercard-type="user" data-hovercard-url="/users/JJHelly/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://urldefense.com/v3/__https://github.com/JJHelly__;!!Mih3wA!FHIGtRkF2o1Oh0lJL-jvSugEBRgh36PpR8sZ5kEQ9FSfzNKPXjiFtpYiDNJsq_3gtvnNMBa1naSofVjtMKLcUhrR$" ***@***.***</a> just to confirm you
        checked this in the INP? And the node names are case sensitive
        too</p>
      <p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">—<br>
        Reply to this email directly, <a href="https://urldefense.com/v3/__https://github.com/pyswmm/pyswmm/discussions/532*discussioncomment-9552104__;Iw!!Mih3wA!FHIGtRkF2o1Oh0lJL-jvSugEBRgh36PpR8sZ5kEQ9FSfzNKPXjiFtpYiDNJsq_3gtvnNMBa1naSofVjtMFxvnrnJ$" moz-do-not-send="true">view it on GitHub</a>, or <a href="https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AZ3YXQDXOK5NOQPSAXXLM33ZD7CAPAVCNFSM6AAAAABICW7OXCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TKNJSGEYDI__;!!Mih3wA!FHIGtRkF2o1Oh0lJL-jvSugEBRgh36PpR8sZ5kEQ9FSfzNKPXjiFtpYiDNJsq_3gtvnNMBa1naSofVjtMGuSiFU9$" moz-do-not-send="true">unsubscribe</a>.<br>
        You are receiving this because you were mentioned.<img src="https://github.com/notifications/beacon/AZ3YXQE52TQTL7ZHYVEWG6LZD7CAPA5CNFSM6AAAAABICW7OXCWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3TUL5UWJTQASHAOQ.gif" alt="" moz-do-not-send="true" width="1" height="1"><span style="color: transparent; font-size: 0; display: none; visibility: hidden; overflow: hidden; opacity: 0; width: 0; height: 0; max-width: 0; max-height: 0; mso-hide: all">Message
          ID: <span>&lt;pyswmm/pyswmm/repo-discussions/532/comments/9552104</span><span>@</span><span>github</span><span>.</span><span>com&gt;</span></span></p>
      <script type="application/ld+json">[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/pyswmm/pyswmm/discussions/532#discussioncomment-9552104",
"url": "https://github.com/pyswmm/pyswmm/discussions/532#discussioncomment-9552104",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]</script>
    </blockquote>
    <br>
    <pre class="moz-signature" cols="72">-- 
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography 
<a class="moz-txt-link-freetext" href="https://www.sdsc.edu/~hellyj">https://www.sdsc.edu/~hellyj</a> / 808 205 9882 / 760 8408660</pre>
  </body>
</html>

--------------SB4Sw5z6Wu3SRkcbfjCps8Lp--

**Comment by bemcdonnell** [comment]
@JJHelly which platform Are you on?

**Comment by JJHelly** [comment]
OSX Sonoma 14.5 (Intel)

J.

On 5/24/24 14:54, Bryant E. McDonnell wrote:
>
> @JJHelly 
> <https://urldefense.com/v3/__https://github.com/JJHelly__;!!Mih3wA!GGOLHLZVSj3j13wKysqna9K0sVHOi9dtJh89L__45Nmm6RRMETuVn8OxwKGl5ucX1d8bSZxfiRheHRzzDYGo49h5$> 
> which platform Are you on?
>
> —
> Reply to this email directly, view it on GitHub 
> <https://urldefense.com/v3/__https://github.com/pyswmm/pyswmm/discussions/532*discussioncomment-9552437__;Iw!!Mih3wA!GGOLHLZVSj3j13wKysqna9K0sVHOi9dtJh89L__45Nmm6RRMETuVn8OxwKGl5ucX1d8bSZxfiRheHRzzDUPYM9OB$>, 
> or unsubscribe 
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AZ3YXQH3FPRJTKHHMDR5LUDZD7OLFAVCNFSM6AAAAABICW7OXCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TKNJSGQZTO__;!!Mih3wA!GGOLHLZVSj3j13wKysqna9K0sVHOi9dtJh89L__45Nmm6RRMETuVn8OxwKGl5ucX1d8bSZxfiRheHRzzDeC0mIlQ$>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

-- 
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj  / 808 205 9882 / 760 8408660

--------------zXKkWVgMNziwLtzq90l7G9mc
Content-Type: text/html; charset=UTF-8
Content-Transfer-Encoding: 8bit

<!DOCTYPE html><html><head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
  </head>
  <body>
    OSX Sonoma 14.5 (Intel)<br>
    <br>
    J.<br>
    <br>
    <div class="moz-cite-prefix">On 5/24/24 14:54, Bryant E. McDonnell
      wrote:<br>
    </div>
    <blockquote type="cite" ***@***.***">
      
      <p dir="auto"><a class="user-mention notranslate" data-hovercard-type="user" data-hovercard-url="/users/JJHelly/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://urldefense.com/v3/__https://github.com/JJHelly__;!!Mih3wA!GGOLHLZVSj3j13wKysqna9K0sVHOi9dtJh89L__45Nmm6RRMETuVn8OxwKGl5ucX1d8bSZxfiRheHRzzDYGo49h5$" ***@***.***</a> which platform Are you on?</p>
      <p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">—<br>
        Reply to this email directly, <a href="https://urldefense.com/v3/__https://github.com/pyswmm/pyswmm/discussions/532*discussioncomment-9552437__;Iw!!Mih3wA!GGOLHLZVSj3j13wKysqna9K0sVHOi9dtJh89L__45Nmm6RRMETuVn8OxwKGl5ucX1d8bSZxfiRheHRzzDUPYM9OB$" moz-do-not-send="true">view it on GitHub</a>, or <a href="https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AZ3YXQH3FPRJTKHHMDR5LUDZD7OLFAVCNFSM6AAAAABICW7OXCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TKNJSGQZTO__;!!Mih3wA!GGOLHLZVSj3j13wKysqna9K0sVHOi9dtJh89L__45Nmm6RRMETuVn8OxwKGl5ucX1d8bSZxfiRheHRzzDeC0mIlQ$" moz-do-not-send="true">unsubscribe</a>.<br>
        You are receiving this because you were mentioned.<img src="https://github.com/notifications/beacon/AZ3YXQGJE7F2U3BVUMGJK5DZD7OLFA5CNFSM6AAAAABICW7OXCWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3TUL5UWJTQASHBDK.gif" alt="" moz-do-not-send="true" width="1" height="1"><span style="color: transparent; font-size: 0; display: none; visibility: hidden; overflow: hidden; opacity: 0; width: 0; height: 0; max-width: 0; max-height: 0; mso-hide: all">Message
          ID: <span>&lt;pyswmm/pyswmm/repo-discussions/532/comments/9552437</span><span>@</span><span>github</span><span>.</span><span>com&gt;</span></span></p>
      <script type="application/ld+json">[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/pyswmm/pyswmm/discussions/532#discussioncomment-9552437",
"url": "https://github.com/pyswmm/pyswmm/discussions/532#discussioncomment-9552437",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]</script>
    </blockquote>
    <br>
    <pre class="moz-signature" cols="72">-- 
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography 
<a class="moz-txt-link-freetext" href="https://www.sdsc.edu/~hellyj">https://www.sdsc.edu/~hellyj</a> / 808 205 9882 / 760 8408660</pre>
  </body>
</html>

--------------zXKkWVgMNziwLtzq90l7G9mc--
